service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java

```diff
@@ -57,6 +78,82 @@ public RecordResponse updateRecord(
         });
   }
 
+  public WdsRecordResponseDetails getRecords(
+      String recordType, List<String> recordIds, BearerToken userToken) {
+
+    int batches = (recordIds.size() / wdsServerConfiguration.queryWindowSize()) + 1;
+    WdsRecordResponseDetails responseDetails =
+        new WdsRecordResponseDetails(new ArrayList<>(), Map.of());
+
+    for (int i = 0; i < batches; i++) {
+      int startIndex = i * wdsServerConfiguration.queryWindowSize();
+      int endIndex = Math.min((i + 1) * wdsServerConfiguration.queryWindowSize(), recordIds.size());
+
+      List<String> batch = recordIds.subList(startIndex, endIndex);
+
+      WdsRecordResponseDetails batchResponse = getRecordsBatch(recordType, batch, userToken);
+      responseDetails = responseDetails.addAll(batchResponse);
+    }
+
+    return responseDetails;
+  }
+
+  private WdsRecordResponseDetails getRecordsBatch(
+      String recordType, List<String> recordIds, BearerToken userToken) {
+
+    try {
+      RecordQueryResponse queryResponse =
+          queryRecords(
+              new SearchRequest()
+                  .filter(new SearchFilter().ids(recordIds))
```

Comment:
When this new filter was released, AJ team had mentioned that it is possible some of WDS apps in prod might not have been updated to have the new search filter for querying records as all the apps might not be updated. But it has been a while since this new version was released. Do you think it would be worth checking with AJ team to see if all existing WDS apps might have the new search filter?

Reply:
They had also mentioned that there is a `capabilities` API that can be used to figure out which features are available in a particular WDS version.

Reply:
+1... if we can determine that the WDS apps have all been updated in prod, this LGTM, otherwise I wonder if we should build in a call to this `capabilities` API to decide which call to make